qemu/tests/pci_bridge.py

```diff
@@ -0,0 +1,308 @@
+    """
+    pci_bridges = params.objects("pci_controllers")
+
+    fmt_list = params.objects("disk_driver")
+    for i in range(image_num):
+        image = "stg%s" % i
+        params["images"] = ' '.join([params["images"], image])
+        params["image_name_%s" % image] = "images/%s" % image
+        params["image_size_%s" % image] = params["data_image_size"]
+        params["force_create_image_%s" % image] = "yes"
+        params["remove_image_%s" % image] = "yes"
+        params["blk_extra_params_%s" % image] = "serial=TARGET_DISK%s" % i
+        if params.get("need_hotplug_unplug", "no") == "no":
+            if i >= len(fmt_list):
+                params["drive_format_%s" % image] = "virtio"
+            else:
+                params["drive_format_%s" % image] = fmt_list[i]
+            d_format = params["drive_format_%s" % image]
+            if d_format == "scsi-hd" and params["drive_format"] == "scsi-hd":
+                params["drive_bus_%s" % image] = 1
+            if d_format == "usb2":
+                params["usbs"] += " ehci"
+                params["usb_type_ehci"] = "usb-ehci"
+                content = "usbc_pci_bus_ehci"
+            else:
+                content = "disk_pci_bus_%s" % image
+            if pci_bridge_num == 1:
+                params[content] = pci_bridges[0]
+            else:
+                index = i % pci_bridge_num
+                params[content] = pci_bridges[index]
+        else:
+            params["boot_drive_%s" % image] = "no"
+        device_num = device_num + 1
+
+    return device_num
+
+
+def prepare_nics(test, params, pci_bridge_num, device_num=0):
+    """
+    Prepare nics attached to pci-bridge for guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param pci_bridge_num: VM pci-bridge number
+    :param device_num: Prepared device number
+    """
+    pci_bridges = params.objects("pci_controllers")
+    nic = params.objects("nics")[0]
+
+    if params.get("need_hotplug_unplug", "no") == "no":
+        if pci_bridge_num == 1:
+            if device_num >= 31:
+                test.fail("There are already %d devices on %s"
+                          % (device_num, pci_bridges[0]))
+            params["nic_pci_bus_%s" % nic] = pci_bridges[0]
+        else:
+            index = device_num % pci_bridge_num
+            params["nic_pci_bus_%s" % nic] = pci_bridges[index]
+        device_num += 1
+
+    return device_num
+
+
+def disk_hotplug(test, params, vm, session, image_name,
+                 drive_format, parent_bus):
+    """
+    Hotplug new disk.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: VM object
+    :param session: VM session
+    :param image: Name of the new disk
+    :param drive_format: Drive subsystem type (virtio, scsi, usb2)
+    :param parent_bus: Bus(es), in which this device is plugged in
+    """
+    def check_usb_in_guest():
```

Comment:
So all usb2 related handling could be removed either.

Reply:
Agree, also note that usb controller `usb-ehci` not available for ppc.
```
USB devices:
name "nec-usb-xhci", bus PCI
name "pci-ohci", bus PCI, desc "Apple USB Controller"
name "qemu-xhci", bus PCI
```

Reply:
If I'm not mistaken, avocado-framework/avocado-vt#1958 forbids using usb2 as the drive format of system disk.
In my case, I need to test various devices connected to pci-bridge work well, including usb-storage.
Considering usb-ehci is not available for ppc, I change to use nec-usb-xhci controller instead.
Could you please review this PR again? Thanks! @PaulYuuu

Reply:
@ybduan `nec-usb-xhci` work well on ppc.
But I hit other problem
```
FAIL: Could not find the usb device serial:[TARGET_DISK2]
```
Will take time to debug it.

Reply:
After I sent `device_add`, it will be deleted after a few seconds.
```
{"timestamp": {"seconds": 1558413543, "microseconds": 617977}, "event": "DEVICE_DELETED", "data": {"path": "/machine/peripheral/stg2/stg2.0/legacy[0]"}}
{"timestamp": {"seconds": 1558413543, "microseconds": 618049}, "event": "DEVICE_DELETED", "data": {"device": "stg2", "path": "/machine/peripheral/stg2"}}
{"timestamp": {"seconds": 1558413543, "microseconds": 625771}, "event": "DEVICE_DELETED", "data": {"device": "usb3", "path": "/machine/peripheral/usb3"}}
```
So guest cannot get its info.

Reply:
@micai20171009 @ngu-niny Could you help to confirm if this is a product issue(usb-storage/block)?

Reply:
> @micai20171009 @ngu-niny Could you help to confirm if this is a product issue(usb-storage/block)?
I boot guest with (-device pci-bridge,id=pci_bridge0,bus=pci.0,addr=0x3,chassis_nr=1  )
then:
{"execute":"device_add", "arguments":{"driver":"nec-usb-xhci", "id":"ehci1", "bus":"pci_bridge0", "addr":3}}
<micai> {"return": {}}
<micai> {"timestamp": {"seconds": 1559102179, "microseconds": 488954}, "event": "DEVICE_DELETED", "data": {"device": "ehci1", "path": "/machine/peripheral/ehci1"}}
<micai> {"timestamp": {"seconds": 1559102188, "microseconds": 518001}, "event": "RTC_CHANGE", "data": {"offset": 1}}

This event occurs when the usb-controlled bus is assigned to a pci-bridge. Therefore, we need to confirm with the development whether special parameter Settings are required.

Reply:
Since this issue can be reproduce manually, so I don't think the ppc issue should be a blocker for this auto patch.  We are confirming this issue and will will file product bug if it is. cc @ybduan @vivianQizhu